botorch/acquisition/joint_entropy_search.py

```diff
@@ -107,5 +285,64 @@ def forward(self, X: Tensor) -> Tensor:
             A `batch_shape`-dim Tensor of acquisition values at the given design
             points `X`.
         """
+        tkwargs = {
+            "dtype": self.optimal_outputs.dtype,
+            "device": self.optimal_outputs.device,
+        }
+
+        initial_posterior = self.initial_model.posterior(X, observation_noise=True)
+
+        add_term = 0.5 * (1 + torch.log(torch.ones(1, **tkwargs) * 2 * pi))
+        batch_shape = X.shape[:-2]
+        sample_dim = len(batch_shape)
+        # We DISREGARD the additional constant term.
+        initial_entropy = 0.5 * torch.logdet(initial_posterior.mvn.covariance_matrix)
+
+        # initial_entropy of shape batch_size or batch_size x num_models if FBGP
+        # first need to unsqueeze the sample dim (first after batch dim) and then the two last
+        initial_entropy = (
+            initial_entropy.unsqueeze(sample_dim).unsqueeze(-1).unsqueeze(-1)
+        )
+
+        CLAMP_LB = torch.finfo(tkwargs["dtype"]).eps
+        # Compute the mixture mean and variance
+        posterior_m = self.conditional_model.posterior(
+            X.unsqueeze(MCMC_DIM), observation_noise=True
+        )
+        noiseless_var = self.conditional_model.posterior(
+            X.unsqueeze(MCMC_DIM), observation_noise=False
```

Comment:
In general this will do a lot of repeated work - for GP models the `observation_noise=True` arg will just add the likelihood noise to the posterior.